src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -243,35 +238,7 @@ private void setMinMaxTerm(ByteBuffer term)
 
     private ByteComparable asComparableBytes(ByteBuffer input)
     {
-        return isLiteral ? version -> terminated(ByteSource.of(input, version))
-                         : version -> index.termType().asComparableBytes(input, version);
-    }
-
-    private ByteComparable decode(ByteComparable term)
-    {
-        return isLiteral ? version -> ByteSourceInverse.unescape(ByteSource.peekable(term.asComparableBytes(version)))
-                         : term;
-    }
-
-    private ByteSource terminated(ByteSource src)
-    {
-        return new ByteSource()
-        {
-            boolean done = false;
-
-            @Override
-            public int next()
-            {
-                if (done)
-                    return END_OF_STREAM;
-                int n = src.next();
-                if (n != END_OF_STREAM)
-                    return n;
-
-                done = true;
-                return ByteSource.TERMINATOR;
-            }
-        };
+        return version -> index.termType().asComparableBytes(input, version);
```

Comment:
So the idea here is that this was never actually necessary for single strings, as the purpose of `ByteSource.TERMINATOR` was to make sequences of `ByteSource`s prefix-free?